core/domain/subtopic_page_services.py

```diff
@@ -0,0 +1,195 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.]
+
+"""Commands for operations on subtopic pages, and related models."""
+
+from core.domain import subtopic_page_domain
+from core.platform import models
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+datastore_services = models.Registry.import_datastore_services()
+memcache_services = models.Registry.import_memcache_services()
+
+
+def get_subtopic_page_from_model(subtopic_page_model):
+    """Returns a domain object for an SubtopicPage given a subtopic page model.
+
+    Args:
+        subtopic_page_model: SubtopicPageModel.
+
+    Returns:
+        SubtopicPage.
+    """
+    return subtopic_page_domain.SubtopicPage(
+        subtopic_page_model.id, subtopic_page_model.topic_id,
+        subtopic_page_model.html_data, subtopic_page_model.language_code,
+        subtopic_page_model.version
+    )
+
+
+def get_subtopic_page_by_id(topic_id, subtopic_id, strict=True):
+    """Returns a domain object representing a topic.
+
+    Args:
+        topic_id: str. ID of the topic that the subtopic is a part of.
+        subtopic_id: int. The id of the subtopic.
+        strict: bool. Whether to fail noisily if no subtopic page with the given
+            id exists in the datastore.
+
+    Returns:
+        SubtopicPage or None. The domain object representing a subtopic page
+            with the given id, or None if it does not exist.
+    """
+    subtopic_page_id = subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
+        topic_id, subtopic_id)
+    subtopic_page_model = topic_models.SubtopicPageModel.get(
+        subtopic_page_id, strict=strict)
+    if subtopic_page_model:
+        subtopic_page = get_subtopic_page_from_model(subtopic_page_model)
+        return subtopic_page
+    else:
+        return None
+
+
+def get_all_subtopic_pages_in_topic(topic_id, include_deleted=False):
```

Comment:
This is interesting -- why do we not retrieve the topic, get the subtopics in it, and fetch the corresponding subtopic pages? In what context would we need the deleted ones?

Reply:
I just added that, if they are needed in the future. Currently, yeah, deleted ones are not needed. Also, we added the topic_id field for this reason, right? That's why I used the query() to get all the subtopic pages related to a topic. Should it be done the other way, to search individual subtopic pages? As that might take more time.

Reply:
Done, modified to not use this function. Though, should I remove the function itself, as it might be useful later on?

Reply:
Remove any functions that are not being used, otherwise they are dead code and hard to maintain.